Hakawai/HKWMentionsPluginV2.m

```diff
@@ -885,16 +884,31 @@ - (void)highlightMentionIfNeededForCursorLocationV2:(NSUInteger)cursorLocation {
 // TODO: Remove text view from call
 // JIRA: POST-14031
 - (BOOL)textView:(UITextView *)textView shouldChangeTextInRange:(NSRange)range replacementText:(NSString *)text {
-    // Remove any mentions selections, since those will go away upon any insertion/deletion
-    [self toggleMentionsFormattingIfNeededAtRange:self.currentlySelectedMentionRange selected:NO];
-    self.currentlySelectedMentionRange = NSMakeRange(NSNotFound, 0);
+    BOOL returnValue = YES;
 
-    // In simple refactor, we only focus on deletions in order to allow for personalization/deletions of mentions
+    // In simple refactor, we only focus on insertions and deletions in order to allow for personalization/deletions/bleaching of mentions.
     if (text.length == 0 && range.length == 1) {
-        return [self handleCharacterDeletionAtLocation:range.location];
+        [self toggleMentionsFormattingIfNeededAtRange:self.currentlySelectedMentionRange selected:NO];
+        self.currentlySelectedMentionRange = NSMakeRange(NSNotFound, 0);
+        returnValue = [self handleCharacterDeletionAtLocation:range.location];
```

Comment:
should we rename this function? it doesn't suggest it returns a bool

Reply:
shouldAllowCharacterDeletionAtLocation